src/main/kotlin/net/lumalyte/lg/infrastructure/services/PartyServiceBukkit.kt

```
@@ -412,6 +414,43 @@ class PartyServiceBukkit(
+                null
+            } else {
+                val updatedParty = party.copy(guildIds = remainingGuilds)
+                if (partyRepository.update(updatedParty)) {
+                    logger.info("Guild $guildId removed from party $partyId (system)")
+                    updatedParty
+                } else {
+                    logger.error("Failed to update party $partyId after system guild removal")
+                    null
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)